### PR TITLE
refactor: cilium config debug to datapathDebug

### LINKED ISSUE #21462: Setting Debug to disabled doesn't stop debug logging
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

When disabling debug logging with `cilium config debug=false` the Debug option in `cilium config output` shows up as `Disabled`, but `debug` level log messages are still being written to output.

```
##### Read-write configurations #####
ConntrackAccounting               : Enabled
ConntrackLocal                    : Disabled
Debug                             : Disabled
(...)
```

### Cilium Version

v1.12.2

I think this also happens in Cilium v1.10 , so it's probably a long-standing bug.

### Kernel Version

I don't think it's relevant.

### Kubernetes Version

1.21

### Sysdump

_No response_

### Relevant log output

```shell
022-09-27T18:25:33.900883947Z level=debug msg="Logging flow record" code=200 header="map[Accept-Encoding:[identity,gzip] Content-Type:[application/grpc] Grpc-Accept-Encoding:[identity,deflate,gzip] X-Envoy-Upstream-Service-Time:[2] X-Request-Id:[7357ab47-d7b7-4a25-b59a-9ea3f57d3d86]]" message= method=POST protocol=HTTP/2 subsys=proxy-logger type=Response url="http://recommendationservice:8080/hipstershop.RecommendationService/ListRecommendations" verdict=Forwarded
2022-09-27T18:25:33.900892758Z level=debug msg="[[C23] stream closed: 0" subsys=envoy-http2 threadID=753
2022-09-27T18:25:33.900899442Z level=debug msg="[[C145649] stream closed: 0" subsys=envoy-http2 threadID=753
2022-09-27T18:25:34.026347319Z level=debug msg="[flushing stats" subsys=envoy-main threadID=713
```

### Anything else?

It should be pretty easy to reproduce, just run `cilium config debug=true`, `cilium config debug=false` and `cilium config` followed by checking the logs output.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by husnialhamdani**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

Refactor `debug` to `datapathDebug` in cilium config as its the correct intention

Fixes: #21462

### CODE REVIEW

File: pkg/option/runtime_options.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ const (
 	PolicyTracing        = "PolicyTracing"
 	ConntrackAccounting  = "ConntrackAccounting"
 	ConntrackLocal       = "ConntrackLocal"
-	Debug                = "Debug"
+	DatapathDebug        = "DatapathDebug"
```
Comment (tklauser) [inline pkg/option/runtime_options.go]: I don't think this const should be renamed, this is the name of the global debug option also infuencing log level, cf. https://github.com/cilium/cilium/blob/1a0553ccaf8ac26a05add34355ae931dd71f4158/daemon/cmd/daemon.go#L1310-L1312

**Review by tklauser — CHANGES_REQUESTED** [review]

File: pkg/option/runtime_options.go (hunk 1 above)
Comment (husnialhamdani) [reply pkg/option/runtime_options.go]: Shall I revert that back for the `runtime_options.go` ?

Comment (tklauser) [reply pkg/option/runtime_options.go]: Yes, please only rename the option when concerning the datapath debug flag. The global daemon debug flag should remain as is.

File: pkg/option/runtime_options.go [hunk 2]
```diff
@@ -35,9 +35,9 @@ var (
 		Requires:    nil,
 	}
 
-	specDebug = Option{
-		Define:      "DEBUG",
-		Description: "Enable debugging trace statements",
+	specDatapathDebug = Option{
+		Define:      "DATAPATH_DEBUG",
+		Description: "Enable debugging trace statements for datapath",
```
Comment (husnialhamdani) [inline pkg/option/runtime_options.go]: Do I need to revert this too?

**Pull request closed without merging**